packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -3668,6 +3668,30 @@ describe('<Autocomplete />', () => {
       expect(textbox).toHaveFocus();
     });
 
+    it('shows input caret when focusing input after chip navigation', () => {
+      const view = render(
+        <Autocomplete
+          defaultValue="two"
+          options={['one', 'two']}
+          renderInput={(params) => <TextField {...params} autoFocus />}
+          renderValue={(value, getItemProps) => {
+            return <Chip label={value} {...getItemProps()} />;
+          }}
+        />,
+      );
+
+      const textbox = screen.getByRole('combobox');
+      const chip = view.container.querySelector(`.${chipClasses.root}`);
+
+      fireEvent.keyDown(textbox, { key: 'ArrowLeft' });
+      expect(chip).toHaveFocus();
+
+      fireEvent.click(textbox);
+
+      expect(textbox).toHaveFocus();
+      expect(chip).not.toHaveFocus();
+    });
```

Comment:
This test doesn't fail if I remove the below logic, which means it is not testing what's required. 

Also, I noticed the relevant issue only occurs when `multiple`, not when a single custom tag is rendered. Why is that the case?